pkg/geo/geogen/geogen.go

```diff
@@ -323,6 +323,14 @@ func RandomGeometry(rng *rand.Rand, srid geopb.SRID) geo.Geometry {
 	return RandomGeometryWithLayout(rng, srid, geom.NoLayout)
 }
 
+// RandomGeometryWithShapeType generates a random Geometry
+// with the given SRID and ShapeType.
+func RandomGeometryWithShapeType(
+	rng *rand.Rand, srid geopb.SRID, shapeType geopb.ShapeType,
+) geo.Geometry {
+	return RandomGeometryWithLayout(rng, srid, geo.ShapeTypeToLayout(shapeType))
```

Comment:
we'll need to actually obey the shape here, e.g. `LINESTRINGM` here by only generating linestrings.

Can you also test this? Something like:

```
typ := types.MakeGeometry(geopb.ShapeType_LINESTRINGM, 4326)
rand := rowenc.RandDatum(... typ ...)
require.NoError(t, geo.SpatialObjectFitsColumnMetadata(4326, geopb.ShapeType_LINESTRINGM)
```